src/service/locator.cpp

```diff
@@ -87,8 +87,14 @@ class locator_t::remote_t:
     virtual
    ~remote_t() {
         for(auto it = active.begin(); it != active.end(); ++it) {
-            parent->m_gateway->cleanup(uuid, *it);
+            if(!parent->m_gateway->cleanup(uuid, *it)) tuple::invoke(*it,
```

Comment:
It's actually `parent->m_protocol[std::get<0>(*it)].erase(std::get<1>(*it));`, not extremely readable, I must say.